### ISSUE TITLE
[REQ] Provide Increment/Decrement preferences for Speed/Incline

### ISSUE THREAD

**Issue by hoveeman** [cagnulein/qdomyos-zwift#618]
**Is your feature request related to a problem? Please describe.**
The + and - for speed and incline in QZ control the treadmill in different increments. Incline is 0.5 increase/decrease and Speed is 0.3 increase/decrease

**Describe the solution you'd like**
I'd like an option in the settings to choose what increment amount I'd like for + and - to use when controlling QZ. I'd like an independent setting for both Incline and Speed.

An example would be:
Incline 0.5 
Speed 0.5
or
Incline 1
Speed 0.1 

A numeric entry box would work for this feature so the user can enter what value they want. A drop down could also work if all values are available.

**Describe alternatives you've considered**
You could also have a Low/Medium/High option for both incline/speed where these could be the values
Low = 0.1
Medium = 0.5
High = 1

**Additional context**
This was referenced in this comment
https://github.com/cagnulein/qdomyos-zwift/issues/592#issuecomment-1021210406

**Comment by cagnulein** [comment]
@hoveeman in your idea, are these settings needs to be converted in miles? i mean if you set the UI in miles, these settings should be converted also that unit?

**Comment by hoveeman** [comment]
@cagnulein, yeah, that would make sense. I'd want the increment to match my desired preference.

So if I have Miles enabled, I would expect .1 to equal .1 mi/h, and .5 to equal .5 mi/h.

**Comment by cagnulein** [comment]
@hoveeman done! send me as usual an email to remember me to send you the beta on iOS i presume

**Comment by hoveeman** [comment]
@cagnulein email sent! Thank you!

**Comment by hoveeman** [comment]
I was able to test this today. Apologies for the delay. I liked the option in the setting to choose the exact amount for the increment and decrement. Overall it worked very well. I set the speed 1.0 increment. It would go up and down in 1 increments. Occasionally though I would adjust it from the treadmill to like 4.3 and then hit - and it would drop it 4 or 3.8 or something like that. Expected would have been 3.3. Most of the time this scenario would work, but occasionally it would not. I could not find a pattern. 

I thought I had a debug log, but I must have had debug disabled. I will test again and upload a debug and give you accurate numbers.

**Comment by hoveeman** [comment]
[debug-Tue_Feb_8_06_55_29_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8033749/debug-Tue_Feb_8_06_55_29_2022.log)
I found the debug file from yesterday. Sorry about that. Let me know if this helps or is confusing. I can try and set up a detailed test and write down the speed numbers and buttons pressed if that would be helpful.

I was flipping back and forth between pressing the physical buttons on the treadmill and then pressing the + and - in QZ.

**Comment by cagnulein** [comment]
@hoveeman just to understand the problem is this one?

    "Current Speed: 5.95"
    "writing speed 7.55" <-- you're pressing the plus button to go to 7.55 km/h
    "Current Speed: 7.55" <-- the treadmill set the speed correctly 
    "Current Speed: 7.55" <-- still fine
    "Current Speed: 7.55" <-- still fine
    "Current Speed: 7.4" <-- why is going down? are you doing something here?
    "Current Speed: 7.4"
    "Current Speed: 7.4"
    "Current Speed: 7.4"
    "Current Speed: 7.08"
    "Current Speed: 7.08"
    "Current Speed: 7.08"
    "Current Speed: 7.08

Maybe your treadmill can't set all the fractions of the speed? Was it working good with the default value 0.5 km/h? maybe it's an issue with the decimal of the miles to km/h conversion? Could you try to use the km/h instead of miles?

**Comment by hoveeman** [comment]
I'm sorry... I couldn't remember what was going on.... I recorded a video today of the issue and the debug log attached. Hopefully this will be more helpful than me trying to explain it... LOL 

It should be easy to see where it occurs with the elapsed tile on the video.

At Elapsed time for the workout 8:27 you can see it in the video and the log. 

Video:
https://youtu.be/M1Y-3JHY0Ss

[debug-Sat_Feb_12_14_25_47_2022.log](https://github.com/cagnulein/qdomyos-zwift/files/8054519/debug-Sat_Feb_12_14_25_47_2022.log)

**Comment by cagnulein** [comment]
@hoveeman the main issue about this it's about a request that some users did sometimes ago [#470 ](https://github.com/cagnulein/qdomyos-zwift/issues/470#issuecomment-952628426) along with the miles/km conversion.
Maybe the best way to solve this is to create a new settings that will disable the alignment to 0.5/1.0/1.5 thing leaving free to add a single mile even if the rounding in KM isn't perfect. What do you think? I hope it makes sense to you :)

**Comment by hoveeman** [comment]
@cagnulein That makes sense! Thanks for sharing the link to that comment. Helps me understand everything better.

**Comment by cagnulein** [comment]
@hoveeman the fix will be in the 2.10.37 ;)

**Comment by cagnulein** [comment]
@hoveeman i sent you the beta if you want to try this out (i don't remember if i had already sent)

**Comment by cagnulein** [comment]
yeah you're right @hoveeman let's wait for this :)

**Comment by cagnulein** [comment]
@hoveeman were you be able to test this at the end?

**Comment by hoveeman** [comment]
@cagnulein I haven't tried it yet. I got distracted by the iPad disconnecting issues I forgot to test this. I'll try and test it asap and report back.

**Comment by hoveeman** [comment]
@cagnulein I gave it a try today and it worked as expected! I tried switching between changing speeds with the treadmill buttons and also QZ. It worked properly every time.

**Comment by nieko-nera** [comment]
Is it treadmill specific setting? It doesn't work with Kingsmith K12 treadmill.

**Comment by cagnulein** [comment]
@nieko-nera no it should work with all of the treadmill but the treadmill has to support this. send me a debug log in case

**Comment by cagnulein** [comment]
fixed @nieko-nera ! it will be in the next one! Thanks!
